### DISCUSSION TITLE
options column

### DISCUSSION THREAD

**Discussion by FelicienLL** [metrumresearchgroup/mrgsolve#877] [General]
Hello Kyle, 
When a text is passed between square braquets with the `@annotated` tag, it is stored in the `options` column in the details tab.  
``` r
code1 <- "
$CMT @annotated
CENT : Comp (mg) [Hello !]"
mod1 <- mcode("mod1", code1)
#> Building mod1 ...
#> done.

as.list(mod1)$details
#> $data
#>   block name descr unit options
#> 1   CMT CENT  Comp   mg Hello !
#> 
#> $embedded
#> [1] TRUE
```
My question is: is it useful for something in particular inside mrgsolve ? Or is it safe to play with it for, let's say, defining default administration / observation compartments for instance ? 
Many thanks in advance
Félicien

**Comment by kylebaron** [comment]
I believe safe to play; haven't touched that in a long time, but I put that in just in case you want to pass something / anything along.  I'll take a look at how that works to see if any gotchas that might be important.

**Comment by kylebaron** [comment]
``` r
x <- "CL : 1 : hey there (mg) [foo, bar, yeah]"
mrgsolve:::parse_annot_line(x)
#> $name
#> [1] "CL"
#> 
#> $value
#> [1] "1"
#> 
#> $unit
#> [1] "mg"
#> 
#> $options
#> [1] "foo, bar, yeah"
#> 
#> $descr
#> [1] "hey there"
```

<sup>Created on 2021-10-18 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
